src/Aligent/AnnouncementBundle/Layout/DataProvider/AnnouncementDataProvider.php

```diff
@@ -120,75 +105,75 @@ public function getDisplayStatus(): bool
     /**
      * Returns the date config value for the provided date config key
      * @param string $dateKey - Date config key
-     * @return string
+     * @return Carbon|null
      */
-    private function getAlertBlockDateConfig(string $dateKey): string
+    protected function getAlertBlockDateConfig(string $dateKey): ?Carbon
     {
         // get config parameters
-        $website = $this->websiteManager->getCurrentWebsite();
-        $dateField = $this->configManager->get(
-            Configuration::getConfigKeyByName($dateKey),
-            false,
-            false,
-            $website
-        );
-        return $dateField ? $dateField->format('d-m-Y') : '';
+        $dateField = $this->getConfiguration($dateKey);
+
+        if (!$dateField instanceof \DateTime) {
+            return null;
+        }
+
+        return $this->toSystemTimeZone(new Carbon($dateField));
+    }
+
+    protected function toSystemTimeZone(Carbon $carbon): Carbon
+    {
+        $timeZone = new DateTimeZone($this->localeSettings->getTimeZone());
+        return $carbon->tz($timeZone);
     }
 
     /**
      * Checks if the announcement should be displayed based on the configured Start and End Dates
      * @return bool
      */
-    private function checkAlertBlockDatesRange(): bool
+    protected function checkAlertBlockDatesRange(): bool
     {
-        $startDateFieldString = $this->getAlertBlockDateConfig(Configuration::ALERT_BLOCK_DATE_START);
-        $endDateFieldString = $this->getAlertBlockDateConfig(Configuration::ALERT_BLOCK_DATE_END);
+        $startDate = $this->getAlertBlockDateConfig(Configuration::ALERT_BLOCK_DATE_START);
+        if ($startDate) {
+            $startDate = $startDate->startOfDay();
+        }
+        $endDate = $this->getAlertBlockDateConfig(Configuration::ALERT_BLOCK_DATE_END);
+        if ($endDate) {
+            $endDate = $endDate->endOfDay();
+        }
 
-        // convert date to one time zone
-        $timeZone = new DateTimeZone($this->localeSettings->getTimeZone());
-        $today = Carbon::now()->tz($timeZone);
-        $startDate = new Carbon($startDateFieldString, $timeZone);
-        $endDate = (new Carbon($endDateFieldString, $timeZone))->endOfDay();
-
-        // compare dates and return display status
```

Comment:
I've wanted to refactor this long `if/elseif/else` statement for ages but had to wait until I had solid test coverage of all the scenarios before rewriting it to ensure there were no regressions. 

There was a scenario where having an end date but no start date would cause a fatal error, this should be resolved now (ie `null->lessThanOrEqualTo($today);`)